### DISCUSSION TITLE
Are you able to use Next.js 13's `app` directory without Turbopack?

### DISCUSSION THREAD

**Discussion by orels1** [vercel/next.js#46628] [Help] [answered]
I'm building a little virtual conference website using the `app` directory, and I've ended up hitting a bit of a roadblock.

A couple of weeks back I faced [this issue](https://github.com/vercel/next.js/issues/45371) which is currently tracked, but have not been resolved yet. Tl;dr on that one is that I am unable to use cookies/headers with Turbopack, because it errors out with `Error: Invariant: Method expects to have requestAsyncStorage, none available`.

Googling around the docs and discussions don't seem to have a clear answer, but is there a way to use the app directory without turbopack to avoid that issue? 13.2 just came out and it adds very nice new routes to replace the old API handling, but I need cookies/headers for my supabase setup to work fully.

Currently, I'm just using a workaround to make it work for unauthorized calls, but that means I am unable to handle any authorized data.

I am not passing `--turbo` anywhere, but still hitting the error, so I assume it just enables automatically if you use the app dir. So if anyone can tell me if/how I can switch to webpack for now, while the issue is being resolved, I would appreciate it!
Thanks 🙏

-------
thx to @ciruz for the pointers

tl;dr - it does not use turbopack by default, but the issue I'm facing was incorrectly marked as turbopack related. The headers do not seem to work correctly even with the default nextjs build, so I'm going to wait for the issue to be closed before trying again

**Comment by ciruz** [comment, marked as answer]
Just this week I finished a next13 project using TypeScript and the `/app` - directory and multiple packages like `redux`, `react-query`, `tailwind`, `leaflet`, `framer-motion`, `react-hook-form` , `fontawesome`, with Server Components and Client Components. **Without turbopack.**

So it's perfectly possible.

Check your `package.json`, is there a --turbo flag?

```jsx
//package.json with turbo
"dev": "next dev --turbo",
```

```jsx
//package.json without turbo
"dev": "next dev",
```

**Reply by orels1** [reply]
I dont have a turbo flag, here's a snippet of my package json

```json
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
```

So, then I assume I am not supposed to be using turbopack, but in that case, it means that using `import { headers, cookies } from 'next/headers'` (issue I mention) is broken for everything, not just turbopack

**Reply by orels1** [reply]
![image](https://user-images.githubusercontent.com/3798928/222226949-a14f98eb-e35f-432a-9e69-7f76726ca3ca.png)

**Reply by orels1** [reply]
Maybe I should just wait for that to be fixed and re-check again, I just assumed this was turbopack related because the person who filed the issue marked it as such, and using/not using the --turbo flag didn't change anything

**Reply by ciruz** [reply]
Do you use the latest next.js canary version?

**Reply by orels1** [reply]
I was not, but I just updated and it has the same issue. So it seems like I'll continue waiting. Either way, thanks for pointing things out, I was totally under the impression that turbopack is being used, while it isn't!

**Reply by ciruz** [reply]
Yeah, when you use Turbopack you get another message in your console.

If I remember correctly, something like

```
TURBOPACK

Thanks for testing ... alpha Version ...
```
